### PR TITLE
import_sstpb: new interface to deny import requests

### PR DESCRIPTION
**Pull request by YuJuncen**

### CODE REVIEW

File: proto/import_sstpb.proto [hunk 1]
```diff
@@ -63,6 +63,27 @@ service ImportSST {
 
     // ClearFiles clear applied file after restore succeed.
     rpc ClearFiles(ClearRequest) returns (ClearResponse) {}
+
+    // Temporarily disable ingest / download / write for data listeners don't support catching import data.
+    rpc DenyImportRPC(DenyImportRPCRequest) returns (DenyImportRPCResponse) {}
```
Comment (disksing) [inline proto/import_sstpb.proto:68]: How about `DisableImport` or `PauseImport`?

File: proto/import_sstpb.proto [hunk 2]
```diff
@@ -63,6 +63,27 @@ service ImportSST {
...
+
+}
+
+message DenyImportRPCRequest {
+    // whether to deny new imports.
+    bool should_deny_imports = 1;
+
+    // whether the duration.
```
Comment (BornChanger) [inline proto/import_sstpb.proto:76]:
```suggestion
    // the duration of import service suspension
```

File: proto/import_sstpb.proto (hunk 1 above)
Comment (BornChanger) [inline proto/import_sstpb.proto:68]:
```suggestion
    rpc SuspendImportRPC(SuspendImportRPCRequest) returns (SuspendImportRPCResponse) {}
```

File: proto/import_sstpb.proto [hunk 3]
```diff
@@ -63,6 +63,27 @@ service ImportSST {
 
     // ClearFiles clear applied file after restore succeed.
     rpc ClearFiles(ClearRequest) returns (ClearResponse) {}
+
+    // Temporarily disable ingest / download / write for data listeners don't support catching import data.
```
Comment (BornChanger) [inline proto/import_sstpb.proto:67]:
```suggestion
    // Suspend ingest / download / write for data listeners don't support catching import data.
```

File: proto/import_sstpb.proto [hunk 4]
```diff
@@ -63,6 +63,27 @@ service ImportSST {
...
+}
+
+message DenyImportRPCRequest {
+    // whether to deny new imports.
+    bool should_deny_imports = 1;
+
+    // whether the duration.
+    // when should_deny_imports is false,
+    // this won't take effect.
+    uint64 duration_secs = 3;
```
Comment (BornChanger) [inline proto/import_sstpb.proto:79]:
```suggestion
    uint64 duration_in_secs = 3;
```

File: proto/import_sstpb.proto [hunk 5]
```diff
@@ -63,6 +63,27 @@ service ImportSST {
...
+    // The identifier for the caller.
+    string caller = 4;
+}
+
+message DenyImportRPCResponse {
```
Comment (BornChanger) [inline proto/import_sstpb.proto:84]:
```suggestion
message SuspendImportRPCResponse {
```

File: proto/import_sstpb.proto [hunk 6]
```diff
@@ -63,6 +63,27 @@ service ImportSST {
...
+    // The last state before this RPC.
+    bool already_denied_imports = 1;
```
Comment (BornChanger) [inline proto/import_sstpb.proto:86]:
```suggestion
    bool already_suspended = 1;
```

File: proto/import_sstpb.proto [hunk 7]
```diff
@@ -63,6 +63,27 @@ service ImportSST {
 
     // ClearFiles clear applied file after restore succeed.
     rpc ClearFiles(ClearRequest) returns (ClearResponse) {}
+
+    // Temporarily disable ingest / download / write for data listeners don't support catching import data.
+    rpc DenyImportRPC(DenyImportRPCRequest) returns (DenyImportRPCResponse) {}
+
+}
+
+message DenyImportRPCRequest {
```
Comment (BornChanger) [inline proto/import_sstpb.proto:72]:
```suggestion
message SuspendImportRPCRequest {
```

File: proto/import_sstpb.proto [hunk 8]
```diff
@@ -63,6 +63,27 @@ service ImportSST {
...
+    // whether to deny new imports.
+    bool should_deny_imports = 1;
```
Comment (BornChanger) [inline proto/import_sstpb.proto:74]:
```suggestion
    // whether to suspend new imports.
    bool should_suspend_imports = 1;
```

File: proto/import_sstpb.proto (hunk 1 above)
Comment (BornChanger) [reply proto/import_sstpb.proto:68]: Checking from https://github.com/pingcap/tidb/pull/47001/files, this rpc is more like a switch function.

File: proto/import_sstpb.proto [hunk 9]
```diff
@@ -63,6 +63,27 @@ service ImportSST {
 
     // ClearFiles clear applied file after restore succeed.
     rpc ClearFiles(ClearRequest) returns (ClearResponse) {}
+
+    // Suspend ingest / download / write for data listeners don't support catching import data.
```
Comment (3pointer) [inline proto/import_sstpb.proto:67]: ingest only

**Review by 3pointer — APPROVED** [review]
rest LGTM

File: proto/import_sstpb.proto [hunk 10]
```diff
@@ -63,6 +63,27 @@ service ImportSST {
...
+}
+
+message SuspendImportRPCRequest {
+    // whether to suspend new imports.
+    bool should_suspend_imports = 1;
+
+    // the duration of import service suspension
+    // when should_deny_imports is false,
+    // this won't take effect.
+    uint64 duration_in_secs = 3;
```
Comment (BornChanger) [inline proto/import_sstpb.proto:79]: why index 2 is skipped?

**Review by BornChanger — APPROVED** [review]

File: proto/import_kvpb.proto [hunk 11]
```diff
@@ -54,6 +54,7 @@ service ImportKV {
     rpc GetMetrics(GetMetricsRequest) returns (GetMetricsResponse) {}
 }
 
+
```
Comment (overvenus) [inline proto/import_kvpb.proto:57]:
```suggestion
```

**Review by overvenus — APPROVED** [review]

**Pull request merged**